diff --git a/projects/rocprofiler-systems/cmake/DyninstTBB.cmake b/projects/rocprofiler-systems/cmake/DyninstTBB.cmake
index 6c004e67..77c5df66 100644
--- a/projects/rocprofiler-systems/cmake/DyninstTBB.cmake
+++ b/projects/rocprofiler-systems/cmake/DyninstTBB.cmake
@@ -153,7 +153,7 @@ else()
     set(TBB_LIBRARY "${TBB_ROOT_DIR}/lib")
     set(TBB_INCLUDE_DIR "${TBB_ROOT_DIR}/include")

-    find_package(TBB ${TBB_MIN_VERSION} COMPONENTS ${_tbb_components})
+    find_package(TBB ${TBB_MIN_VERSION} CONFIG COMPONENTS ${_tbb_components})

     if(NOT TBB_FOUND)
         if(STERILE_BUILD)
diff --git a/projects/rocprofiler-systems/cmake/Packages.cmake b/projects/rocprofiler-systems/cmake/Packages.cmake
index ac1f4d75..36749afa 100644
--- a/projects/rocprofiler-systems/cmake/Packages.cmake
+++ b/projects/rocprofiler-systems/cmake/Packages.cmake
@@ -562,6 +562,14 @@ if(ROCPROFSYS_BUILD_DYNINST)
 else()
     # Find Boost before finding Dyninst
     find_package(Boost)
+
+    find_package(
+        Dyninst
+        ${rocprofiler_systems_FIND_QUIETLY}
+        REQUIRED
+        COMPONENTS dyninstAPI parseAPI instructionAPI symtabAPI
+    )
+
     if(NOT TARGET Dyninst::Boost_headers)
         add_library(Dyninst::Boost_headers INTERFACE IMPORTED)
         target_include_directories(
@@ -571,13 +579,6 @@ else()
         )
     endif()

-    find_package(
-        Dyninst
-        ${rocprofiler_systems_FIND_QUIETLY}
-        REQUIRED
-        COMPONENTS dyninstAPI parseAPI instructionAPI symtabAPI
-    )
-
     if(TARGET Dyninst::Dyninst) # updated Dyninst CMake system was found
         target_link_libraries(rocprofiler-systems-dyninst INTERFACE Dyninst::Dyninst)
     else() # updated Dyninst CMake system was not found
